primitiveFTPd/src/org/primftpd/filesystem/VirtualFtpFile.java

```diff
@@ -44,7 +44,7 @@ public String getClientIp() {
     @Override
     public boolean move(FtpFile target) {
         logger.trace("move()");
-        return delegate != null && ((FtpFile) delegate).move(target);
+        return delegate != null && ((FtpFile) delegate).move((FtpFile) ((VirtualFtpFile) target).delegate);
```

Comment:
Cast to FtpFile should be removed and only be casted to VirtualFtpFile

Reply:
The FtpFile cast is for the delegate! ;) The delegate is AbstractFile, we have to cast it to something, and we call FtpFile's move(), and that uses FtpFile as argument. So I think this is correct as is.

Reply:
Ah, now I see. The important piece is to pass target.delegate. Therefore all the casts are necessary.